### ISSUE TITLE
AxisLimitsChanged event: the emitted `RenderDetails` contains old information

### ISSUE THREAD

**Issue by BrianAtZetica** [ScottPlot/ScottPlot#3508]
**Question:** Why are the RenderDetails one render behind when implementing an AxisChanged event handler? Is this a WPF only issue?

**ScottPlot Version:** current github main branch

**Code Sample:** 
In the following code sample (modified from the WPF sandbox from the repo) I have subscribed to the AxisLimitsCHanged event with my AxisChanged handler. In my handler I set the title of the plot to show the "AxisLimits" from the RenderDetails object.

```cs
using System.Linq;
using System.Windows;
using ScottPlot;

namespace Sandbox.WPF;

public partial class MainWindow : Window
{
    public MainWindow()
    {
        InitializeComponent();
        WpfPlot1.Plot.RenderManager.AxisLimitsChanged += AxisChanged;
        double[] Xs = Enumerable.Range(1, 5000).Select(i => (double)i / 1000).ToArray();
        var signal = WpfPlot1.Plot.Add.SignalXY(Xs, Generate.Sin(count: 5000, oscillations: 4));

        void AxisChanged(object sender, RenderDetails e)
        {
            var changedPlot = (ScottPlot.Plot)sender;
            var newLimits = e.AxisLimits;
            WpfPlot1.Plot.Title($"New Y limits: {newLimits.Bottom:F1}, {newLimits.Top:F1} New X limits: {newLimits.Left:F1}, {newLimits.Right:F1}");
        }
    }
}

```

When first rendered, the AxisChanged event must have been called during initialisation and plotting of the data, and the title has the correct limits:
<img width="372" alt="image" src="https://github.com/ScottPlot/ScottPlot/assets/119825052/4fd09c3b-4771-461d-bafb-745b7799ae79">

But if I zoom out once with the mousewheel the axis limits in the title are old (note: I had to screengrab this, because a copy to clip board from the right click context menu would force another render and the saved image title is then correct!):
<img width="379" alt="image" src="https://github.com/ScottPlot/ScottPlot/assets/119825052/0c907657-df41-4346-9031-469ae6ec715b">

If I then reset zoom with mouse wheel click
the graph resets correctly, but the axis title shows the previous zoom extent:
<img width="374" alt="image" src="https://github.com/ScottPlot/ScottPlot/assets/119825052/2204584d-dae3-4134-8e3a-8a59b7920b48">

I require a second click of the mouse wheel to force another zoom to extents to see the correct limits in the title:
<img width="372" alt="image" src="https://github.com/ScottPlot/ScottPlot/assets/119825052/c8e6b4d4-43c6-477f-8775-09c0de0a83d4">

**Context to Question**
I intend to update RenderDetails.cs to store the previous render's axis limits

```cs
    /// <summary>
    /// Axis limits of the primary axes for the previous render
    /// </summary>
    public readonly AxisLimits PreviousAxisLimits;
```
Which would be populated inthe constructor:
```cs
        PreviousAxisLimits = lastRender.AxisLimits;
```

This would then allow the nature of the AxisLimitsChange to be established within the AxisLimitsChanged event handler.

**Comment by BrianAtZetica** [comment]
It looks like RenderDetails contains current information afterall, but my attempt to see the latest information by setting the plot title was illinformed. 

The AxisChanged event is happening after the render has occurred, and there is no new render when the title (or xlabel/ylabel) are set.

Should setting the labels automatically trigger a new render, or do we expect user to call their own render?

I've tried adding a call of WpfPlot1.Plot.Render() after setting the title, but even though I can see the renderCount is incremented, the title is still not updated.

Which render action would be responsible for setting the title?

```CS
    public static List<IRenderAction> DefaultRenderActions => new()
    {
        new RenderActions.PreRenderLock(),
        new RenderActions.ClearCanvas(),
        new RenderActions.RenderFigureBackground(),
        new RenderActions.ReplaceNullAxesWithDefaults(),
        new RenderActions.AutoScaleUnsetAxes(),
        new RenderActions.ExecutePlottableAxisManagers(),
        new RenderActions.ApplyAxisRulesBeforeLayout(),
        new RenderActions.CalculateLayout(),
        new RenderActions.ApplyAxisRulesAfterLayout(),
        new RenderActions.RegenerateTicks(),
        new RenderActions.RenderStartingEvent(),
        new RenderActions.RenderDataBackground(),
        new RenderActions.RenderGridsBelowPlottables(),
        new RenderActions.RenderPlottables(),
        new RenderActions.RenderGridsAbovePlottables(),
        new RenderActions.RenderLegends(),
        new RenderActions.RenderPanels(),
        new RenderActions.RenderZoomRectangle(),
        new RenderActions.SyncGLPlottables(),
        new RenderActions.RenderPlottablesLast(),
        new RenderActions.RenderBenchmark(),
    };

```

**Comment by swharden** [comment]
resolved in #3525
